### PR TITLE
Add compose 3 parsing, tests

### PR DESCRIPTION
**Pull request by allisaurus**
`make` && `make test` pass

### CODE REVIEW

File: ecs-cli/modules/cli/compose/project/project_parseV3.go [hunk 1]
```diff
@@ -11,20 +17,93 @@ import (
...
+	}
+	return filepath.Dir(fileName), nil
+}
+
+func getLocalEnv() (map[string]string, error) {
+	osEnv := os.Environ()
+	envMap := make(map[string]string)
+
+	for _, envString := range osEnv {
+		if !strings.Contains(envString, "=") {
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3.go]: Hm, would this realistically ever happen? From what I can tell, `os.Environ()` makes a system call to get the env variables on a given system. 

More to the point though, do we actually only care about a particular set of environment variables? I assumed that maybe was the case on reading the error message, which is why validating only on having the string contain an `=` was a bit confusing to me.

File: ecs-cli/modules/cli/compose/project/project_parseV3.go [hunk 2]
```diff
@@ -11,20 +17,93 @@ import (
...
+	wrkDir, err := getWorkingDir(p.ecsContext.ComposeFiles[0])
+	if err != nil {
+		return nil, err
+	}
+
+	// build config details
+	configDetails := types.ConfigDetails{
+		WorkingDir:  wrkDir,
+		ConfigFiles: configFiles,
+		Environment: env,
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3.go]: for some reason I'm getting the heebie-jeebies about this. I realize that this is what docker/cli defines on `ConfigDetails` but unless i'm missing something it seems a little weird that we're copying the customer's whole environment into here?

File: ecs-cli/modules/cli/compose/project/project_parseV3.go [hunk 3]
```diff
@@ -11,20 +17,93 @@ import (
...
-		convertDockerToContainerConfig(service)
+		cCon, err := convertDockerToContainerConfig(service)
+		if err != nil {
+			return nil, err
+		}
+		conConfigs = append(conConfigs, *cCon)
 	}
 
-	return &[]containerconfig.ContainerConfig{}, nil
+	// return container configs
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3.go]: Nit: Unnecessary comment?

File: ecs-cli/modules/cli/compose/project/project_parseV3.go [hunk 4]
```diff
@@ -11,20 +17,93 @@ import (
...
+	return &conConfigs, nil
+}
+
+func convertDockerToContainerConfig(serviceConfig types.ServiceConfig) (*containerconfig.ContainerConfig, error) {
+	//TODO: fully convert ServiceConfig to ContainerConfig
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3.go]: Ah, is this pending something else?

File: ecs-cli/modules/cli/compose/project/project_parseV3.go [hunk 5]
```diff
@@ -11,20 +17,93 @@ import (
...
+	c := &containerconfig.ContainerConfig{
+		Name: serviceConfig.Name,
+	}
+	return c, nil
 }
 
-func convertDockerToContainerConfig(serviceConfig types.ServiceConfig) {
-	return
+func getWorkingDir(fileName string) (string, error) {
+	if strings.Index(fileName, "/") != 0 {
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3.go]: Instead of manually checking the path and concatting the filename with dir name, you can use `filepath.Abs()`:
```
	pwd := filepath.Dir(filename)
	dir, err := filepath.Abs(pwd)
	if err != nil {
                return "", errors.Wrap(err, "I HAZ PORBLEM")
	}
```

File: ecs-cli/modules/cli/compose/project/project_parseV3_test.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	if err != nil {
+		t.Fatal("Unexpected error parsing file: ", err)
+	}
+	assert.Equal(t, numOfServices, len(*conConfigResult))
+}
+
+func TestParseV3WithMultileFiles(t *testing.T) {
+	// set up files
+	numOfServices := 3
+	firstFileString := `version: '3'
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3_test.go]: Nittiest of nits: given you name the files `tmpfile1` and `tmpile2`, perhaps `filestring1` and `filestring2` would help make things less verbose/more consistent?

File: ecs-cli/modules/cli/compose/project/project_parseV3.go [hunk 7]
```diff
@@ -11,20 +17,93 @@ import (
...
+	}
+
+	// build config details
+	configDetails := types.ConfigDetails{
+		WorkingDir:  wrkDir,
+		ConfigFiles: configFiles,
+		Environment: env,
+	}
+
+	// load config from config details
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3.go]: Nit: In generally I like having comments to explain steps of a process, but in this case I feel like the code documents itself pretty well, so could do without the comment. Unless there are more relevant deatils about loading the config that you wanted to make explicit? (same with other code comments in here)

File: ecs-cli/modules/cli/compose/project/project_parseV3.go [hunk 8]
```diff
@@ -11,20 +17,93 @@ import (
...
+		}
+		conConfigs = append(conConfigs, *cCon)
 	}
 
-	return &[]containerconfig.ContainerConfig{}, nil
+	// return container configs
+	return &conConfigs, nil
+}
+
+func convertDockerToContainerConfig(serviceConfig types.ServiceConfig) (*containerconfig.ContainerConfig, error) {
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3.go]: Hm, what do you think about renaming this to simply `convertToContainerConfig`? Not sure I like "Docker" in the method name, and "ServiceConfig" seems too wordy. Given that this method is unexported, I'd say there's probably enough context that saying what you're converting to might be sufficient?

File: ecs-cli/modules/cli/compose/project/project_parseV3.go (hunk 3 above)
Comment (allisaurus) [reply ecs-cli/modules/cli/compose/project/project_parseV3.go]: yep! will remove - the TODO on line 79 already conveys what's left to be done

File: ecs-cli/modules/cli/compose/project/project_parseV3.go (hunk 4 above)
Comment (allisaurus) [reply ecs-cli/modules/cli/compose/project/project_parseV3.go]: this is where the ServiceCnfg --> containerConfig work will happen (next PR)

File: ecs-cli/modules/cli/compose/project/project_parseV3.go (hunk 7 above)
Comment (allisaurus) [reply ecs-cli/modules/cli/compose/project/project_parseV3.go]: I'm ok with removing some of the comments here - they're mostly a holdover from when i was scaffolding out what needed to be done

File: ecs-cli/modules/cli/compose/project/project_parseV3.go (hunk 8 above)
Comment (allisaurus) [reply ecs-cli/modules/cli/compose/project/project_parseV3.go]: agreed! will change

File: ecs-cli/modules/cli/compose/project/project_parseV3.go (hunk 2 above)
Comment (allisaurus) [reply ecs-cli/modules/cli/compose/project/project_parseV3.go]: it looks like is the Env is not supplied, Docker makes sys call to get the LookupEnv:
https://github.com/docker/cli/blob/master/cli/compose/interpolation/interpolation.go#L31
 
Given this, I'm inclined to leave the sys calling to Docker unless we find a compelling reason to set it ourselves.

File: ecs-cli/modules/cli/compose/containerconfig/containerconfig.go [hunk 9]
```diff
@@ -3,6 +3,8 @@ package containerconfig
 // ContainerConfig all compose fields supported by the ecs-cli
 // TODO: finalize fields
 type ContainerConfig struct {
+	Name string
```
[the remark this answers is not in the document]
Comment (allisaurus) [reply ecs-cli/modules/cli/compose/containerconfig/containerconfig.go]: talked offline: this is the result of running `fmt` , diff field blocks indent independently
